sunpy/image/transform.py

```diff
@@ -106,25 +107,35 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
             np.nan_to_num(image).T, rmatrix, offset=shift, order=order,
             mode='constant', cval=missing).T
     else:
-        # Make the rotation matrix 3x3 to include translation of the image
-        skmatrix = np.zeros((3, 3))
-        skmatrix[:2, :2] = rmatrix
-        skmatrix[2, 2] = 1.0
-        skmatrix[:2, 2] = shift
-        tform = skimage.transform.AffineTransform(skmatrix)
-
-        # Transform the image using the skimage function
-        if not np.issubdtype(image.dtype, np.float64):
-            warnings.warn("Input data has been cast to float64.", SunpyUserWarning)
-            adjusted_image = image.astype(np.float64)
-        else:
-            adjusted_image = image.copy()
-        if np.any(np.isnan(adjusted_image)) and order >= 4:
+        tform = _affine_transform_matrix(rmatrix, shift)
+
+        rotated_image = image.copy()
+        # skikit-image expects floating point images to be in the range -1, 1
```

Comment:
```suggestion
        # scikit-image expects floating point images to be in the range 0, 1
```

Reply:
Hmm, perhaps we have to be cleverer to deal with negative values too.